src/sentry/services/hybrid_cloud/app/__init__.py

```diff
@@ -1,15 +1,113 @@
 from __future__ import annotations
 
 import abc
+import datetime
 from dataclasses import dataclass, field
-from typing import List
+from typing import TYPE_CHECKING, Any, List, Mapping, Optional, Protocol, TypedDict
 
 from sentry.constants import SentryAppInstallationStatus
+from sentry.models import SentryApp, SentryAppInstallation
 from sentry.services.hybrid_cloud import InterfaceWithLifecycle, silo_mode_delegation, stubbed
+from sentry.services.hybrid_cloud.filter_query import FilterQueryInterface
 from sentry.silo import SiloMode
 
+if TYPE_CHECKING:
+    from sentry.mediators.external_requests.alert_rule_action_requester import AlertRuleActionResult
+
+
+@dataclass
+class RpcSentryAppService:
+    title: str = ""
+    slug: str = ""
+    service_type: str = "sentry_app"
+
+
+@dataclass
+class RpcSentryAppInstallation:
+    id: int = -1
+    organization_id: int = -1
+    status: int = SentryAppInstallationStatus.PENDING
+    sentry_app: RpcSentryApp = field(default_factory=lambda: RpcSentryApp())
+    date_deleted: Optional[datetime.datetime] = None
+    uuid: str = ""
+
+
+@dataclass
+class RpcSentryAppComponent:
+    uuid: str = ""
+    sentry_app_id: int = -1
+    type: str = ""
+    schema: Mapping[str, Any] = field(default_factory=dict)
+
+
+ApiSentryAppInstallation = RpcSentryAppInstallation
+
+
+@dataclass
+class RpcSentryApp:
+    id: int = -1
+    scope_list: List[str] = field(default_factory=list)
+    application_id: int = -1
+    proxy_user_id: int | None = None  # can be null on deletion.
+    owner_id: int = -1  # relation to an organization
+    name: str = ""
+    slug: str = ""
+    uuid: str = ""
+    events: List[str] = field(default_factory=list)
+
+
+ApiSentryApp = RpcSentryApp
```

Comment:
Do we still need the Api* aliases?

Reply:
I'd like to take them all out in a separate PR, only because of concurrent work that still references them.  Once I flatten my PRs, I will have a final PR to remove any open Api* alias.  I promise you won't I forget!